tasks/add_picklist_values.py

```diff
@@ -0,0 +1,332 @@
+        <label>{label}</label>
+        <picklistValues>
+            <picklist>{field}</picklist>{picklist_values}
+        </picklistValues>
+    </recordTypes>"""
+
+record_type_picklist_value_template = """
+            <values>
+                <fullName>{name}</fullName>
+                <default>{default}</default>
+            </values>"""
+
+class AddPicklistValues(BaseSalesforceApiTask, Deploy):
+    task_options = {
+        "sobject": {
+            "description": "SObject of the picklist field being modified",
+            "required": True,
+        },
+        "field": {
+            "description": "The picklist field being modified",
+            "required": True,
+        },
+        "values": {
+            "description": "A comma-delimited list of the picklist values being added",
+            "required": True,
+        },
+        "recordtypes": {
+            "description": "The record types for which these new picklist values will be available",
+            "required": False,
+        },
+        "sorted": {
+            "description": "If true, sets the entire picklist to be sorted in alphabetical order",
+            "required": False,
+        },
+        "otherlast": {
+            "description": "If true, the Other value (if one exists) will remain at the end of the values",
+            "required": False,
+        }
+    }
+
+    # Adds picklist values to the given field, if they don't already exist.
+    # Optionally adds the picklist values for the specified record types, if the record types exist.
+    # Optionally updates the picklist values to be alphabetical.
+    def _run_task(self):
+        self.api_version = "47.0"
+        sobject = self.options["sobject"]
+        field = self.options["field"]
+        values = self.options["values"].split(',')
+
+        # The Tooling API does not have the namespace or the '__c' suffix in the DeveloperName of the CustomField object
+        # or in the DeveloperName of the EntityDefinition object, so we have to remove it.
+        split_field = field.split('__')
+        if len(split_field) is 3:
+            field_developer_name = split_field[1]
+            namespace = split_field[0]
+        else:
+            field_developer_name = split_field[0]
+            namespace = ""
+
+        split_sobject = sobject.split('__')
+        if len(split_sobject) is 3:
+            sobject_developer_name = split_sobject[1]
+        else:
+            sobject_developer_name = split_sobject[0]
+
+        sobject_tooling_results = self.tooling.query(
+            "SELECT DeveloperName, DurableId "
+            "FROM EntityDefinition "
+            "WHERE NamespacePrefix = '{namespace}' "
+            "AND DeveloperName = '{sobject}'".format(
+                namespace = namespace,
+                sobject = sobject_developer_name
+            )
+        )
+
+        if sobject_tooling_results["size"] is not 1:
+            raise Exception('{} object not found'.format(sobject))
```

Comment:
@jstvz @davisagli For everywhere I'm throwing an exception, am I doing it correctly? (Is there a better way to do this without displaying a large stacktrace to the user?)

I assume it's probably okay since this would only be encountered by a dev or QE when testing that the feature that this task is implementing is working as designed.

Reply:
Don't worry about stack traces -- we are working on changing how CumulusCI deals with them across the board, so that it will not show them by default, but they'll be available after the fact for those who want to debug.

Reply:
I'll leave this up to you but this would be a good place to use an f-string instead of the format method:

    raise Exception(f"{sobject} object not found")